sympy/series/tests/test_order.py

```diff
@@ -334,6 +334,18 @@ def test_issue_6753():
     assert (1 + x**2)**10000*O(x) == O(x)
 
 
+def test_issue_9917():
+    from sympy.abc import n
+    from sympy import O, oo, sin
+
+    assert O(n*sin(n) + 1, (n, oo)) == O(n*sin(n) + 1, (n, oo))
```

Comment:
This line, as well as some of the following, do not actually test the behavior of Order. Same object on both sides of equality, so equality is assured to hold regardless of what that object is. So it holds in current master branch as well. 

Perhaps you meant to test 
```
O(n*sin(n)+1, (n, oo)).expr == n*sin(n) + 1
```
which would indeed verify that the expression returned by Order constructor is what you expect.